### DISCUSSION TITLE
How to adaptively display a large image when first appear

### DISCUSSION THREAD

**Discussion by weiminghuaa** [ebarellar/LFColoringBook#3] [Q&A]
This is a great project to implementation of a Coloring Book app. I first tried to implement it with PencilKit, but encountered a lot of problems. This project helped me, thank you very much. It’s just that some images are very big in my project, how can I solve it，let the image to adapt to a fixed size at first time diaplay?

![bird](https://user-images.githubusercontent.com/2723525/116809542-0c3e5d80-ab71-11eb-9ab9-4a40a2fb1b2c.png)

![IMG_B45391B92683-1](https://user-images.githubusercontent.com/2723525/116809538-06487c80-ab71-11eb-94a8-7a5730e3f3ba.jpeg)

**Comment by ebarellar** [comment]
Thank you for finding interest in the project. Yeah, I don’t think PencilKit is flexible enough for something like this yet

Wow, are your images taking too long to be processed?

Anyway, answering your question, you would need to adjust the zoomScale of the UIScrollView in the ViewController. In the createColoringBookView method after adding the new coloring book view to the scrollView calculate how much zoom do you need to start with based on the size of the new coloring book view

**Reply by weiminghuaa** [reply]
Thany you for reply! I have resolved this problem. I found that adjust the zoomScale after set anchor constraint is useless. It's need set before anchor constraint. 

I pack all the mask into the project main bundle, this way avoid taking too long to be processed.
